### PR TITLE
add notice about anonymous telemetry data collection

### PR DESCRIPTION
**Pull request by abernix**
Add a message indicating our (intent to) collection of anonymized usage data that we'll use to do product development (e.g., counts, etc.) along with a privacy document which is similar to that [of Rover's doc](https://www.apollographql.com/docs/rover/privacy/).

### CODE REVIEW

File: apollo-router/src/executable.rs [hunk 1]
```diff
@@ -238,7 +238,15 @@ impl Executable {
...
-        let apollo_router_msg = format!("Apollo Router v{} // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)", std::env!("CARGO_PKG_VERSION"));
+        let value = std::env::var("APOLLO_TELEMETRY_DISABLED").ok();
+        let is_telemetry_disabled = value.is_some();
+        let apollo_telemetry_msg = if is_telemetry_disabled {
+            "Telemetry has been disabled.".to_string()
+        } else {
+            "Telemetry is enabled. To disable, set APOLLO_TELEMETRY_DISABLED=1".to_string()
+        };
+
+        let apollo_router_msg = format!("Apollo Router v{} // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)\n{}", apollo_telemetry_msg, std::env!("CARGO_PKG_VERSION"));
```
Comment (Geal) [inline apollo-router/src/executable.rs:249]:
```suggestion
        let apollo_router_msg = format!("Apollo Router v{} // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)\n{}", std::env!("CARGO_PKG_VERSION"), apollo_telemetry_msg);
```

Comment (o0Ignition0o) [reply apollo-router/src/executable.rs:249]: they're backwards indeed!

File: apollo-router/src/executable.rs [hunk 2]
```diff
@@ -258,7 +258,14 @@ impl Executable {
                 .unwrap_or_else(|| Configuration::builder().build().into()),
         };
 
-        let apollo_router_msg = format!("Apollo Router v{} // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)", std::env!("CARGO_PKG_VERSION"));
+        let is_telemetry_disabled = std::env::var("APOLLO_TELEMETRY_DISABLED").ok().is_some();
+        let apollo_telemetry_msg = if is_telemetry_disabled {
+            "Anonymous usage data was disabled via APOLLO_TELEMETRY_DISABLED=1.".to_string()
+        } else {
+            "Anonymous usage data is gathered to inform Apollo product development.  For info, see https://go.apollo.dev/o/privacy.".to_string()
```
Comment (abernix) [inline apollo-router/src/executable.rs:265]:
```suggestion
            "Anonymous usage data is gathered to inform Apollo product development.  See https://go.apollo.dev/o/privacy for more info.".to_string()
```

File: docs/source/privacy.mdx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+title: Privacy and data collection
+---
+
+By default, Apollo Router collects some anonymous usage data to help us improve the product.
```
Comment (StephenBarlow) [inline docs/source/privacy.mdx:5]:
```suggestion
By default, the Apollo Router collects some anonymous usage data to help us improve the product.
```

File: docs/source/privacy.mdx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+**To opt out of data collection,** set the `APOLLO_TELEMETRY_DISABLED` environment variable to `1` in each environment where you run the Apollo Router.
+
+> Apollo Router doesn't collect _any_ personally identifiable information (such as API keys, graph names, or file paths). For more information on how Apollo collects and uses this data, see [our privacy policy](https://www.apollographql.com/Apollo-Privacy-Policy.pdf).
```
Comment (StephenBarlow) [inline docs/source/privacy.mdx:9]:
```suggestion
> The Apollo Router doesn't collect _any_ personally identifiable information (such as API keys, graph names, or file paths). For more information on how Apollo collects and uses this data, see [our privacy policy](https://www.apollographql.com/Apollo-Privacy-Policy.pdf).
```

File: docs/source/privacy.mdx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+## Collected data
+
+Unless you opt out, Apollo Router reports the following data:
```
Comment (StephenBarlow) [inline docs/source/privacy.mdx:13]:
```suggestion
Unless you opt out, the Apollo Router reports the following data:
```

**Review by StephenBarlow — APPROVED** [review]
Just adding some "the" to match rest of docs

File: apollo-router/src/executable.rs [hunk 6]
```diff
@@ -258,7 +258,14 @@ impl Executable {
                 .unwrap_or_else(|| Configuration::builder().build().into()),
         };
 
-        let apollo_router_msg = format!("Apollo Router v{} // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)", std::env!("CARGO_PKG_VERSION"));
+        let is_telemetry_disabled = std::env::var("APOLLO_TELEMETRY_DISABLED").ok().is_some();
+        let apollo_telemetry_msg = if is_telemetry_disabled {
+            "Anonymous usage data was disabled via APOLLO_TELEMETRY_DISABLED=1.".to_string()
+        } else {
+            "Anonymous usage data is gathered to inform Apollo product development.  See https://go.apollo.dev/o/privacy for more info.".to_string()
```
[the remark this answers is not in the document]
Comment (abernix) [reply apollo-router/src/executable.rs:265]: Thanks for the rest of the review @StephenBarlow.  Any thoughts on this bit? (Trying to keep it short and sweet and informative.)

[the remark this answers is not in the document]
Comment (StephenBarlow) [reply apollo-router/src/executable.rs:265]: This looks ok to me for brevity's sake!

File: docs/source/privacy.mdx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+**To opt out of data collection,** set the `APOLLO_TELEMETRY_DISABLED` environment variable to `1` in each environment where you run the Apollo Router.
+
+> The Apollo Router doesn't collect _any_ personally identifiable information (such as API keys, graph names, or file paths). For more information on how Apollo collects and uses this data, see [our privacy policy](https://www.apollographql.com/Apollo-Privacy-Policy.pdf).
+
+## Collected data
+
+Unless you opt out, the Apollo Router reports the following data:
+
+- The execution command that was run _(excluding any identifiable arguments such as file-system paths or profile names)_
```
Comment (prasek) [inline docs/source/privacy.mdx:15]:
```suggestion
- The execution command that was run _(excluding any identifiable arguments such as file-system paths or profile names)_
- The `router --schema` properties referenced _(excluding any identifiable property values in the `router` config such as file-system paths or URLs)_
```

**Pull request merged**